salt/tests/unit/runners/test_metalk8s_saltutil.py

```diff
@@ -0,0 +1,87 @@
+import os.path
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from parameterized import param, parameterized
+from salt.exceptions import CommandExecutionError
+import yaml
+
+from _runners import metalk8s_saltutil
+
+from tests.unit import mixins
+from tests.unit import utils
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), "files", "test_metalk8s_saltutil.yaml"
+)
+with open(YAML_TESTS_FILE) as fd:
+    YAML_TESTS_CASES = yaml.safe_load(fd)
+
+
+class Metalk8sSaltutilTestCase(TestCase, mixins.LoaderModuleMockMixin):
+    """
+    TestCase for `metalk8s_saltutil` runner
+    """
+
+    loader_module = metalk8s_saltutil
+
+    def test_sync_auth(self):
+        """
+        Tests the return of `sync_auth` function
+        """
+        sync_mock = MagicMock()
+
+        with patch("salt.utils.extmods.sync", sync_mock):
+            metalk8s_saltutil.sync_auth(saltenv="my-salt-env")
+
+            sync_mock.assert_called_once()
+            assert "auth" in sync_mock.call_args[0]
+            self.assertDictContainsSubset(
+                {"saltenv": "my-salt-env"}, sync_mock.call_args[1]
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["wait_minions"])
+    def test_wait_minions(
+        self, result, ping_ret=True, is_running_ret=False, raises=False
+    ):
+        """
+        Tests the return of `wait_minions` function
+        """
+        minion_id = "my-minion"
+
+        def cmd_mock(_tgt, command, **_kwargs):
+            ret = None
+            if command == "test.ping":
+                ret = ping_ret
+            elif command == "saltutil.is_running":
+                ret = is_running_ret
+
+            if ret is None:
+                raise CommandExecutionError("An ErrOr Occurred")
```

Comment:
Why not go one step further and instead use 1337 speak! :)
```
4n 3rr0r 0ccurr3d
```
No no, don't do that. :)